container.go

```diff
@@ -83,85 +83,72 @@ func (c *Container) Invoke(t interface{}) error {
 	return nil
 }
 
-// Provide an object in the Container
+// Provide registers all the provided args in the Container
 //
 // The provided argument must be a function that accepts its dependencies as
 // arguments and returns one or more results, which must be a pointer type, map, slice or an array.
 // The function may optionally return an error as the last argument.
-func (c *Container) Provide(t interface{}) error {
-	ctype := reflect.TypeOf(t)
-	switch ctype.Kind() {
-	case reflect.Func:
-		switch ctype.NumOut() {
-		case 0:
-			return errReturnCount
-		case 1:
-			objType := ctype.Out(0)
-			if objType.Kind() != reflect.Ptr && objType.Kind() != reflect.Interface {
-				return errReturnKind
+func (c *Container) Provide(types ...interface{}) error {
+	for _, t := range types {
+		ctype := reflect.TypeOf(t)
+		switch ctype.Kind() {
+		case reflect.Func:
+			switch ctype.NumOut() {
+			case 0:
+				return errReturnCount
+			case 1:
+				objType := ctype.Out(0)
+				if objType.Kind() != reflect.Ptr && objType.Kind() != reflect.Interface {
+					return errReturnKind
+				}
 			}
+			if err := c.Graph.InsertConstructor(t); err != nil {
+				return err
+			}
+		case reflect.Slice, reflect.Array, reflect.Map, reflect.Ptr:
```

Comment:
Array is more a value type not a pointer like Slice/Map/Interface, not sure if it should be here.